code/go/0chain.net/core/common/rate_limiter.go

```diff
@@ -1,159 +1,162 @@
+		DefaultExpirationTTL: time.Minute * 5,
+	})
+	generalRL.SetIgnoreURL(true)
+	generalRL.SetIPLookups(defaultIPLookups)
 
-	fileRateLimit.RateLimit = true
-	fileRateLimit.Limiter = tollbooth.NewLimiter(fileRateLimit.RequestsPerSecond, &limiter.ExpirableOptions{DefaultExpirationTTL: time.Hour}).
-		SetIPLookups([]string{}). // IP not required
-		SetIgnoreURL(false)       // URL Path contains allocation
 }
 
 //ConfigRateLimits - configure the rate limits
 func ConfigRateLimits() {
-	initUserRateLimiter(viper.GetFloat64("handlers.rate_limit"))
-	initFileRateLimiter(viper.GetFloat64("handlers.file_rate_limit"))
-}
-
-func NewGRPCRateLimiter() *GRPCRateLimiter {
-	userRl := viper.GetFloat64("handlers.rate_limit")
-
-	return &GRPCRateLimiter{rl.New(int(userRl))}
-}
+	tokenExpirettl := viper.GetDuration("rate_limiters.default_token_expire_duration")
+	isProxy := viper.GetBool("rate_limiters.proxy")
+
+	if tokenExpirettl > 0 {
+		commitRL.SetTokenBucketExpirationTTL(tokenExpirettl)
+		fileRL.SetTokenBucketExpirationTTL(tokenExpirettl)
+		objectRL.SetTokenBucketExpirationTTL(tokenExpirettl)
+		generalRL.SetTokenBucketExpirationTTL(tokenExpirettl)
+	}
 
-type GRPCRateLimiter struct {
-	rl.Limiter
-}
+	if isProxy {
+		// If blobber is behind some proxy then it is important to put
+		// "X-Forwarded-For" in fron of other parameters.
+		ipLookup := []string{"X-Forwarded-For", "RemoteAddr", "X-Real-IP"}
+		commitRL.SetIPLookups(ipLookup)
+		fileRL.SetIPLookups(ipLookup)
+		objectRL.SetIPLookups(ipLookup)
+		generalRL.SetIPLookups(ipLookup)
+	}
 
-func (r *GRPCRateLimiter) Limit() bool {
-	r.Take()
-	return false
-}
+	cRps := viper.GetFloat64("rate_limiters.commit_rps")
+	fRps := viper.GetFloat64("rate_limiters.file_rps")
+	oRps := viper.GetFloat64("rate_limiters.object_rps")
+	gRps := viper.GetFloat64("rate_limiters.general_rps")
 
-//UserRateLimit - rate limiting for end user handlers
-func UserRateLimit(handler ReqRespHandlerf) ReqRespHandlerf {
-	if !userRateLimit.RateLimit {
-		return handler
+	if cRps > 0 {
+		commitRL.SetMax(cRps)
 	}
-	return func(writer http.ResponseWriter, request *http.Request) {
-		tollbooth.LimitFuncHandler(userRateLimit.Limiter, handler).ServeHTTP(writer, request)
-	}
-}
 
-//UseUserRateLimit - rate limiting for end user handlers
-func UseUserRateLimit(h http.Handler) http.Handler {
-	if !userRateLimit.RateLimit {
-		return h
+	if fRps > 0 {
+		fileRL.SetMax(fRps)
 	}
 
-	return tollbooth.LimitFuncHandler(userRateLimit.Limiter, h.ServeHTTP)
-}
+	if oRps > 0 {
+		objectRL.SetMax(oRps)
+	}
 
-// FileRateLimit is a custom of `tollbooth.LimitHandlerFunc` that uses custom values to build keys for limiting file actions.
-func FileRateLimit(handler ReqRespHandlerf) ReqRespHandlerf {
-	if fileRateLimit == nil || !fileRateLimit.RateLimit {
-		return handler
+	if gRps > 0 {
+		generalRL.SetMax(gRps)
 	}
 
-	lmt := fileRateLimit.Limiter
+}
 
+func RateLimit(handler ReqRespHandlerf, rlType RPS) ReqRespHandlerf {
```

Comment:
Instead of using the `rlType` to different different rate limit and expose those types to users, I would suggest to define different rate limit functions and users can call them directly, they don't need to know the `RPS`, all of these could be private. For example, we can define

```
func RateLimitByCommit(handle ReqRespHandlerf) ReqRespHandlerf {
    return rateLimit(handler, LImitByCommitRL)
}

func RateLimitByFile(handler ReqRespHandlerf) ReqRespHandlerf {
    return rateLimit(handler, LimitByFileRL)
}
...

```

Then in the API handlers, we can use it like:

```go

r.HandleFunc("/v1/file/upload/{allocation}",
		common.RateLimitByFile(common.ToJSONResponse(WithConnection(UploadHandler))))

and
r.HandleFunc("/v1/connection/commit/{allocation}",
common.RateLimitByCommit(common.ToStatusCode(WithStatusConnection(CommitHandler))))

```